contracts/utils/CommonUtils.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import 'forge-std/Test.sol';
+
+import '../../contracts/hts-precompile/KeyHelper.sol';
+
+/// generic test utils
+abstract contract CommonUtils is Test, KeyHelper {
```

Comment:
Q: should everything under `contracts/utils` be under `contracts/mocks/utils`?
That seems to be where it's used and I'm not clear it's providing value to the example contracts.
They seem all focused on test scenarios as they use or import `forge-std/Test.sol`.

Reply:
[I previously suggested](https://github.com/hashgraph/hedera-smart-contracts/pull/236#discussion_r1320654970) moving `contracts/mocks` and `contracts/utils` into `test/foundry`. This is pretty standard in a foundry project, and there shouldn't be any hardhat compile issues since hardhat won't try to compile the mocks which requires `forge-std` and fails since that dependency hasn't been configured in hardhat

Reply:
@mshakeg This seems like a good idea.
In your current PR `contracts/mocks` is all foundry focused but I can see a general use case of `contracts/mocks` soon.
If I missed this before my bad but yeah let's make this move to make it more self contained for foundry.
Thanks

Reply:
Hey @Nana-EC I think I've addressed all of your feedback in this review, also got the tests ci to pass, I just had to re-run all failed jobs a few times, not sure why this is, could be an issue with the local node intermittently not returning the latest and correct state.

Reply:
Thanks @mshakeg 
We've seen a little flakiness with the CI flow but are resolving that.
I just gave a latest review, i think just one more commit and we should be good to merge this.
Thanks for your patience and contributions